File: crates/analytics/src/xata/track.rs
use chrono::{DateTime, Utc};
use serde::Deserialize;

#[derive(Debug, sqlx::FromRow, Deserialize, Clone)]
pub struct Track {
    pub xata_id: String,
    pub title: String,
    pub artist: String,
    pub album_artist: String,
    pub album_art: Option<String>,
    pub album: String,
    pub track_number: i32,
    pub duration: i32,
    pub mb_id: Option<String>,
    pub youtube_link: Option<String>,
    pub spotify_link: Option<String>,
    pub tidal_link: Option<String>,
    pub apple_music_link: Option<String>,
    pub sha256: String,
    pub lyrics: Option<String>,
    pub composer: Option<String>,
    pub genre: Option<String>,
    pub disc_number: i32,
    pub copyright_message: Option<String>,
    pub label: Option<String>,
    pub uri: Option<String>,
    pub artist_uri: Option<String>,
    pub album_uri: Option<String>,
    #[serde(with = "chrono::serde::ts_seconds")]
    pub xata_createdat: DateTime<Utc>,
}

File: remote-ws/lib/remote_ws/schema/track.ex
defmodule RemoteWs.Schema.Track do
  @moduledoc "Mirrors apps/api schema/tracks (the `tracks` table)."
  use Ecto.Schema

  @primary_key {:id, :string, source: :xata_id, autogenerate: false}
  schema "tracks" do
    field :title, :string
    field :artist, :string
    field :album, :string
    field :album_artist, :string
    field :album_art, :string
    field :duration, :integer
    field :sha256, :string
    field :uri, :string
    field :album_uri, :string
    field :artist_uri, :string
  end
end
